### PR TITLE
Allow all numeric types with scale 0 as frame offset

### PR DESCRIPTION
**Pull request by kasiafi**
This change is similar to https://github.com/trinodb/trino/pull/6891, but additionally long decimal is supported.
It required moving correctness checks on the frame offset values out of the operator. This solution is more complicated than https://github.com/trinodb/trino/pull/6891, but it is analogous to how frames of type RANGE are handled.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java [hunk 1]
```diff
@@ -706,13 +706,13 @@ public void testNonBooleanWhereClause()
     @Test
     public void testDistinctAggregations()
     {
-        analyze("SELECT COUNT(DISTINCT a), SUM(a) FROM t1");
```
Comment (martint) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java:709]: Unrelated change

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java [hunk 2]
```diff
@@ -706,13 +706,13 @@ public void testNonBooleanWhereClause()
...
+        analyze("SELECT COUNt(DISTINCT a), SUM(a) FROM t1");
     }
 
     @Test
     public void testMultipleDistinctAggregations()
     {
-        analyze("SELECT COUNT(DISTINCT a), COUNT(DISTINCT b) FROM t1");
```
Comment (martint) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java:715]: Unrelated change

Comment (kasiafi) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java]: fixed

File: core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java [hunk 3]
```diff
@@ -1093,6 +1106,76 @@ private FrameBoundPlanAndSymbols planFrameBound(PlanBuilder subPlan, PlanAndMapp
...
+            String maxBigint = Long.toString(Long.MAX_VALUE);
+            int maxBigintPrecision = maxBigint.length();
+            int actualPrecision = ((DecimalType) offsetType).getPrecision();
+
+            if (actualPrecision < maxBigintPrecision) {
+                offsetToBigint = new Cast(offsetSymbol.toSymbolReference(), toSqlType(BIGINT));
+            }
+            else if (actualPrecision > maxBigintPrecision) {
+                // In case of long overflow, truncate to max bigint
+                offsetToBigint = new GenericLiteral("BIGINT", maxBigint);
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java:1150]: Why truncate? This could theoretically result in incorrect results, no? If so, it'd be better to just fail with a "out of bounds" error.

Comment (kasiafi) [reply core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java:1150]: If the offset value exceeds max bigint, it implies that the frame bound falls beyond partition bound. In such case, there is no error, but the frame bound is set to the partition bound. Passing max bigint as the offset value has the same effect.

Comment (martint) [reply core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java:1150]: ok, please add a comment with that explanation
